### PR TITLE
Add option to disable typing notifications 

### PR DESCRIPTION
**Pull request by schwukas**
Resolves [this riot-im issue](https://github.com/vector-im/riot-web/issues/7283).

Signed-off-by: Lukas Schwarz <ghub@posteo.se>

### CODE REVIEW

**Comment by t3chguy** [comment]
I believe as per the contributing doc it has to be name and e-mail address.

Also your name has to fit within:

> We accept contributions under a legally identifiable name, such as your name on government documentation or common-law names (names claimed by legitimate usage or repute). Unfortunately, we cannot accept anonymous contributions at this time.

**Comment by schwukas** [comment]
That's alright. I fixed it :)

**Comment by schwukas** [comment]
I don't see any reason why this option would be unwanted. I've hear several people wishing for it because the typing notifications are distracting to them. I just find them unnecessary. Since this options defaults to "show typing notifications" anyway, I don't understand the problem of including it. However, this decision is not up to me.

If you find it to be a worthy inclusion just tell me, and I'll gladly add a test.

**Comment by t3chguy** [comment]
I'd say given Tom labelled it as a feature with a priority it seems like having it behind a default-off setting should be fine

File: src/settings/Settings.js [hunk 1]
```diff
@@ -224,6 +224,12 @@ export const SETTINGS = {
         default: true,
         invertedSettingName: 'dontSendTypingNotifications',
     },
+    "showTypingNotifications": {
+        supportedLevels: LEVELS_ACCOUNT_SETTINGS,
+        displayName: _td("Show typing notifications"),
+        default: true,
+        invertedSettingName: 'dontShowTypingNotifications',
```
Comment (turt2live) [inline src/settings/Settings.js]: This shouldn't be here - there's no legacy setting to maintain.

Comment (schwukas) [reply src/settings/Settings.js]: Are you referring solely to line 231? It's a bit unclear from how github marked it.

Comment (turt2live) [reply src/settings/Settings.js]: yes, 231. The `invertedSettingName` is a backwards compatibility thing, but there's nothing to be backwards compatible with here.

**Comment by turt2live** [comment]
> I'd say given Tom labelled it as a feature with a priority it seems like having it behind a default-off setting should be fine

It's also been a year - we should double check.

**Comment by t3chguy** [comment]
image

This is now a thing

**Pull request closed without merging**